exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet2/DrillParquetReader.java

```diff
@@ -179,6 +189,31 @@ private static MessageType getProjection(MessageType schema,
     return projection;
   }
 
+  /**
+   * Get type from the supplied {@code type} corresponding to given {@code segment}.
+   * @param type type to extract field corresponding to segment
+   * @param segment segment which type will be returned
+   * @return type corresponding to the {@code segment} or {@code null} if there is no field found in {@code type}.
+   */
+  private static Type getType(Type type, PathSegment segment) {
+    Type result = null;
+    if (type != null && !type.isPrimitive()) {
+      GroupType groupType = type.asGroupType();
+      if (segment.isNamed()) {
+        String fieldName = segment.getNameSegment().getPath();
+        Optional<Type> foundType = groupType.getFields().stream()
```

Comment:
Is it required to declare `foundType` here, since methods calls chain may be continued?